src/PortingAssistantExtensionServer/Services/AnalysisService.cs

```diff
@@ -61,16 +61,16 @@ public async Task<SolutionAnalysisResult> AssessSolutionAsync(AnalyzeSolutionReq
                 var solutionAnalysisResult = await _client.AnalyzeSolutionAsync(request.solutionFilePath, request.settings);
 
                 Collector.SolutionAssessmentCollect(
```

Comment:
Can we try catch the metric collection separately? From the log it seems error throw from collector, we shouldn't block assessment if only the metric failed.

Reply:
Also let's add metric when exception throw from assessment. if assessment couldn't finish its an critical error should reflect in the availability.

Reply:
and maybe add some null check/default value for fields in the metric to avoid throw exception in the metrics

Reply:
I think I could address concern 1 and 3. You have a good point that it will be impacting our availability, but I don't think that would be in the scope of this PR.